client/my-sites/themes/theme-showcase.jsx

```diff
@@ -125,6 +125,14 @@ class ThemeShowcase extends Component {
 		) {
 			this.scrollToSearchInput();
 		}
+
+		if (
+			! isEqual( prevProps.subjects, this.props.subjects ) ||
+			prevProps.siteCanInstallThemes !== this.props.siteCanInstallThemes
```

Comment:
I think we might still need to check `isJetpackSite` and `isAtomicSite`. How about making a function named `shouldShowMyThemesFilter( { isJetpackSite, isAtomicSite, siteCanInstallThemes } )` to decide whether we should update the filters or not?

BTW, both `tabFilters` and `tabSubjectTermTable` are the member variables, so the component might not be re-rendered after we update their value 😅

Reply:
Thanks for your suggestions. What was I thinking 😂 will fix this!

Reply:
> BTW, both tabFilters and tabSubjectTermTable are the member variables, so the component might not be re-rendered after we update their value 😅

The component is actually rerendered if `subjects` /  `siteCanInstallThemes` changes. However, the rerendered component will be incorrect because `tabFilters` and `tabSubjectTermTable` are not updated. This is the bug.

Reply:
@arthur791004 I modified the logic of this PR to only initialize static data in constructors. Derived data from props are (ultimately) calculated during `render()`. Could you re-check this? Thanks!